common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1013,6 +1014,21 @@ public static UTF8String concatWs(UTF8String separator, UTF8String... inputs) {
     return res;
   }
 
+  public UTF8String[] split_delimiter(UTF8String delimiter, int limit) {
```

Comment:
We can have a private method `UTF8String[] split(String delimiter, int limit)`, and 2 public methods: `UTF8String[] split(UTF8String pattern, int limit)`, `UTF8String[] SqlSplit(UTF8String delimiter, int limit)`. The 2 public methods just call the private method with different delimiter strimg.

Reply:
In the future, we can add a variant of the SQL `split` function that calls the `splitSQL` method here, which adopts the common SQL semantic for the delimiter parameter.